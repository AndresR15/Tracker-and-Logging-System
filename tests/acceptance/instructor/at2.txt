-- at2.txt 
-- Error messages for new tracker and phases
-- comments may occur at the beginning or in the middle of a line
-- type CONTAINER = TUPLE[material: MATERIAL; radioactivity: VALUE]

-- phases sorted by pid
-- containers sorted by cid

-- create a new tracker
-- max. radiation in a phase is 50.0
-- max. radiation in a container is 10.0
new_tracker(39.0, 10.0)

-- error cases for new tracker
new_tracker(-1, 10.0)  -- e2
new_tracker(50.0, -1)  -- e3
new_tracker(1.0, 1.1)  -- e4
new_tracker(0, 1) -- error case

-- add phases pid3 and pid2
new_phase("pid3", "compacting", 2, <<glass, metal, plastic>>)
new_phase("pid2", "assay", 3, <<glass, metal, plastic, liquid>>)

-- error checking on adding new phases
new_phase("", "assay", 3, <<glass, metal, plastic, liquid>>) -- e5
new_phase("pid2", "assay", 3, <<glass, metal, plastic, liquid>>) -- e6
new_phase("pid9", "", 3, <<glass, metal, plastic, liquid>>) -- e5
new_phase("pid9", "junk", 0, <<glass, metal, plastic, liquid>>) -- e7
new_phase("pid9", "junk", 1, <<>>) -- e8

new_phase("pid1", "unpacking", 4, <<glass, metal, plastic, liquid>>)
new_phase("pid4", "storage", 7, <<glass, metal, plastic, liquid>>)
