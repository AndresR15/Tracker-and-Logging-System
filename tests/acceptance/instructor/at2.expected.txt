  state 0 ok
  max_phase_radiation: 0.00, max_container_radiation: 0.00
  phases: pid->name:capacity,count,radiation
  containers: cid->pid->material,radioactivity
->new_tracker(50,10)
  state 1 ok
  max_phase_radiation: 50.00, max_container_radiation: 10.00
  phases: pid->name:capacity,count,radiation
  containers: cid->pid->material,radioactivity
->new_phase("pid2","compacting",2,<<glass, metal, plastic>>)
  state 2 ok
  max_phase_radiation: 50.00, max_container_radiation: 10.00
  phases: pid->name:capacity,count,radiation
    pid2->compacting:2,0,0.00,{glass,metal,plastic}
  containers: cid->pid->material,radioactivity
->new_phase("pid1","unpacking",2,<<glass, metal, plastic, liquid>>)
  state 3 ok
  max_phase_radiation: 50.00, max_container_radiation: 10.00
  phases: pid->name:capacity,count,radiation
    pid1->unpacking:2,0,0.00,{glass,metal,plastic,liquid}
    pid2->compacting:2,0,0.00,{glass,metal,plastic}
  containers: cid->pid->material,radioactivity
->new_container("cid4",[metal, 3],"pid1")
  state 4 ok
  max_phase_radiation: 50.00, max_container_radiation: 10.00
  phases: pid->name:capacity,count,radiation
    pid1->unpacking:2,1,3.00,{glass,metal,plastic,liquid}
    pid2->compacting:2,0,0.00,{glass,metal,plastic}
  containers: cid->pid->material,radioactivity
    cid4->pid1->metal,3.00
->new_container("cid1",[glass, 5.5],"pid1")
  state 5 ok
  max_phase_radiation: 50.00, max_container_radiation: 10.00
  phases: pid->name:capacity,count,radiation
    pid1->unpacking:2,2,8.50,{glass,metal,plastic,liquid}
    pid2->compacting:2,0,0.00,{glass,metal,plastic}
  containers: cid->pid->material,radioactivity
    cid1->pid1->glass,5.50
    cid4->pid1->metal,3.00
->new_container("cid2",[liquid, 0.5],"pid1")
  state 6 e11: this container will exceed phase capacity
->move_container("cid1","pid1","pid2")
  state 7 ok
  max_phase_radiation: 50.00, max_container_radiation: 10.00
  phases: pid->name:capacity,count,radiation
    pid1->unpacking:2,1,3.00,{glass,metal,plastic,liquid}
    pid2->compacting:2,1,5.50,{glass,metal,plastic}
  containers: cid->pid->material,radioactivity
    cid1->pid2->glass,5.50
    cid4->pid1->metal,3.00
->move_container("cid4","pid1","pid2")
  state 8 ok
  max_phase_radiation: 50.00, max_container_radiation: 10.00
  phases: pid->name:capacity,count,radiation
    pid1->unpacking:2,0,0.00,{glass,metal,plastic,liquid}
    pid2->compacting:2,2,8.50,{glass,metal,plastic}
  containers: cid->pid->material,radioactivity
    cid1->pid2->glass,5.50
    cid4->pid2->metal,3.00
->new_container("cid4",[metal, 3],"pid1")
  state 9 e10: this container identifier already in tracker
->new_container("cid1",[glass, 5.5],"pid1")
  state 10 e10: this container identifier already in tracker
->new_container("cid2",[liquid, 0.5],"pid1")
  state 11 ok
  max_phase_radiation: 50.00, max_container_radiation: 10.00
  phases: pid->name:capacity,count,radiation
    pid1->unpacking:2,1,0.50,{glass,metal,plastic,liquid}
    pid2->compacting:2,2,8.50,{glass,metal,plastic}
  containers: cid->pid->material,radioactivity
    cid1->pid2->glass,5.50
    cid2->pid1->liquid,0.50
    cid4->pid2->metal,3.00
->undo
  state 12 (to 10) e10: this container identifier already in tracker
->undo
  state 13 (to 9) e10: this container identifier already in tracker
->undo
  state 14 (to 8) ok
  max_phase_radiation: 50.00, max_container_radiation: 10.00
  phases: pid->name:capacity,count,radiation
    pid1->unpacking:2,0,0.00,{glass,metal,plastic,liquid}
    pid2->compacting:2,2,8.50,{glass,metal,plastic}
  containers: cid->pid->material,radioactivity
    cid1->pid2->glass,5.50
    cid4->pid2->metal,3.00
->undo
  state 15 (to 7) ok
  max_phase_radiation: 50.00, max_container_radiation: 10.00
  phases: pid->name:capacity,count,radiation
    pid1->unpacking:2,1,3.00,{glass,metal,plastic,liquid}
    pid2->compacting:2,1,5.50,{glass,metal,plastic}
  containers: cid->pid->material,radioactivity
    cid1->pid2->glass,5.50
    cid4->pid1->metal,3.00
->undo
  state 16 (to 6) e11: this container will exceed phase capacity
->undo
  state 17 (to 5) ok
  max_phase_radiation: 50.00, max_container_radiation: 10.00
  phases: pid->name:capacity,count,radiation
    pid1->unpacking:2,2,8.50,{glass,metal,plastic,liquid}
    pid2->compacting:2,0,0.00,{glass,metal,plastic}
  containers: cid->pid->material,radioactivity
    cid1->pid1->glass,5.50
    cid4->pid1->metal,3.00
->undo
  state 18 (to 4) ok
  max_phase_radiation: 50.00, max_container_radiation: 10.00
  phases: pid->name:capacity,count,radiation
    pid1->unpacking:2,1,3.00,{glass,metal,plastic,liquid}
    pid2->compacting:2,0,0.00,{glass,metal,plastic}
  containers: cid->pid->material,radioactivity
    cid4->pid1->metal,3.00
->undo
  state 19 (to 3) ok
  max_phase_radiation: 50.00, max_container_radiation: 10.00
  phases: pid->name:capacity,count,radiation
    pid1->unpacking:2,0,0.00,{glass,metal,plastic,liquid}
    pid2->compacting:2,0,0.00,{glass,metal,plastic}
  containers: cid->pid->material,radioactivity
->undo
  state 20 (to 2) ok
  max_phase_radiation: 50.00, max_container_radiation: 10.00
  phases: pid->name:capacity,count,radiation
    pid2->compacting:2,0,0.00,{glass,metal,plastic}
  containers: cid->pid->material,radioactivity
->undo
  state 21 (to 1) ok
  max_phase_radiation: 50.00, max_container_radiation: 10.00
  phases: pid->name:capacity,count,radiation
  containers: cid->pid->material,radioactivity
->undo
  state 22 e19: there is no more to undo
->undo
  state 23 e19: there is no more to undo
->redo
  state 24 (to 2) ok
  max_phase_radiation: 50.00, max_container_radiation: 10.00
  phases: pid->name:capacity,count,radiation
    pid2->compacting:2,0,0.00,{glass,metal,plastic}
  containers: cid->pid->material,radioactivity
->redo
  state 25 (to 3) ok
  max_phase_radiation: 50.00, max_container_radiation: 10.00
  phases: pid->name:capacity,count,radiation
    pid1->unpacking:2,0,0.00,{glass,metal,plastic,liquid}
    pid2->compacting:2,0,0.00,{glass,metal,plastic}
  containers: cid->pid->material,radioactivity
->redo
  state 26 (to 4) ok
  max_phase_radiation: 50.00, max_container_radiation: 10.00
  phases: pid->name:capacity,count,radiation
    pid1->unpacking:2,1,3.00,{glass,metal,plastic,liquid}
    pid2->compacting:2,0,0.00,{glass,metal,plastic}
  containers: cid->pid->material,radioactivity
    cid4->pid1->metal,3.00
->redo
  state 27 (to 5) ok
  max_phase_radiation: 50.00, max_container_radiation: 10.00
  phases: pid->name:capacity,count,radiation
    pid1->unpacking:2,2,8.50,{glass,metal,plastic,liquid}
    pid2->compacting:2,0,0.00,{glass,metal,plastic}
  containers: cid->pid->material,radioactivity
    cid1->pid1->glass,5.50
    cid4->pid1->metal,3.00
->redo
  state 28 (to 6) e11: this container will exceed phase capacity
->redo
  state 29 (to 7) ok
  max_phase_radiation: 50.00, max_container_radiation: 10.00
  phases: pid->name:capacity,count,radiation
    pid1->unpacking:2,1,3.00,{glass,metal,plastic,liquid}
    pid2->compacting:2,1,5.50,{glass,metal,plastic}
  containers: cid->pid->material,radioactivity
    cid1->pid2->glass,5.50
    cid4->pid1->metal,3.00
->redo
  state 30 (to 8) ok
  max_phase_radiation: 50.00, max_container_radiation: 10.00
  phases: pid->name:capacity,count,radiation
    pid1->unpacking:2,0,0.00,{glass,metal,plastic,liquid}
    pid2->compacting:2,2,8.50,{glass,metal,plastic}
  containers: cid->pid->material,radioactivity
    cid1->pid2->glass,5.50
    cid4->pid2->metal,3.00
->redo
  state 31 (to 9) e10: this container identifier already in tracker
->redo
  state 32 (to 10) e10: this container identifier already in tracker
->redo
  state 33 (to 11) ok
  max_phase_radiation: 50.00, max_container_radiation: 10.00
  phases: pid->name:capacity,count,radiation
    pid1->unpacking:2,1,0.50,{glass,metal,plastic,liquid}
    pid2->compacting:2,2,8.50,{glass,metal,plastic}
  containers: cid->pid->material,radioactivity
    cid1->pid2->glass,5.50
    cid2->pid1->liquid,0.50
    cid4->pid2->metal,3.00
->redo
  state 34 e20: there is no more to redo
->redo
  state 35 e20: there is no more to redo
